models/marts/ecommerce/fct_orders.sql

```diff
@@ -0,0 +1,36 @@
+{{
+    config(
+        materialized='table'
+    )
+}}
+
+with 
+orders as (
+    select * from {{ ref('orders_xf') }}
+),
+
+transactions as (
+    select 
+        *,
+        row_number() over (partition by order_id order by 
```

Comment:
I think the more simple version of this is to do a last value window function for the two fields that you are looking to pull from transactions as opposed to row_number and then filtering. it also provides greater flexibility in the future if you want to incorporate first paid date and most recent, etc

Reply:
Transactions have a higher level of granularity than orders, so there needs to be a filtering to ensure there is only one transaction (the latest) per order. Are you suggesting something like this?
```
transactions as (
    select distinct
        
        order_id,
         
        last_value(created_at) over (partition by order_id order by 
            created_at) as paid_at,
            
        last_value(gateway) over (partition by order_id order by 
            created_at) as payment_method
            
    from {{ ref('stg_shopify_transactions') }}
), 
...
```

Reply:
yes! 100%